c/expr/binaryop.cc

```diff
@@ -195,7 +195,7 @@ inline static VT op_div(LT x, RT y) {
   VT vx = static_cast<VT>(x);
   VT vy = static_cast<VT>(y);
   VT res = vx / vy;
-  if ((vx < 0) != (vy < 0) && vx != res * vy) {
+  if (std::is_integral<VT>::value && (vx < 0) != (vy < 0) && vx != res * vy) {
```

Comment:
- What is the reason we need to decrement `res` here?

- Am I correct that `vx` and `vy` could be integer, and `res` float? If yes, then are you sure the condition `vx != res * vy` won't always be true due to floating point arithmetics on the right hand side?

Reply:
This accounts for the difference in division operator between python in C. In C, x/y truncates the result towards 0, whereas in python x/y returns the floor. Thus, in C `(-1)/2 == 0`, while in python `(-1)//2 == -1`.

Reply:
If `res` is float, then `std::is_integral<VT>::value` will be false, and the if condition will be ignored.

Reply:
Ah, yes. With this change it will not pass through.

Btw, what are the cases when we divide integer over integer without casting them to float?

Reply:
Then what is the case you provided the above example for? I mean when we may end up with integers being divided as `(-1) / 2`?

Reply:
In python integer division is `//`. So, `(-1) // 2 == -1`, while `(-1) / 2 == -0.5`.